### ISSUE TITLE
SUGGESTION: Adding Dynamic Touch Screen Joystick in Flare Android Port

### ISSUE THREAD

**Issue by realprincemj** [flareteam/flare-game#838]
[](https://www.youtube.com/watch?v=NdWZq8vdKhs) Here's an example of the joystick, Playing with the current joystick in Flare 1.11 feels old and hard, It's hard to control the character your playing compared to games like Zenonia or MOBA Games but This is just a suggestion.

**Comment by dorkster** [comment]
Interesting idea.

If I'm understanding correctly, the screen is divided into vertical slices for each control. For our configuration, I would imagine the D-pad in the first 50% of the screen with the second 50% divided evenly based on the number of action buttons we have (currently 2). "Evenly" might not be the right approach, since the action buttons should probably be biased to the right edge of the screen. But once these regions are set up, it should be easy to recenter inputs on each touch.

The only conflict I can think of is that the player can currently interact with map objects by tapping on them when using touch controls. This wouldn't work since tapping anywhere on the screen would just move the closest virtual input. Not much of an issue, because we could simply make it behave more like our physical gamepad behavior by adding a virtual button that maps to the "Accept" input.

**Comment by realprincemj** [comment]
> Interesting idea.
> 
> If I'm understanding correctly, the screen is divided into vertical slices for each control. For our configuration, I would imagine the D-pad in the first 50% of the screen with the second 50% divided evenly based on the number of action buttons we have (currently 2). "Evenly" might not be the right approach, since the action buttons should probably be biased to the right edge of the screen. But once these regions are set up, it should be easy to recenter inputs on each touch.
> 
> The only conflict I can think of is that the player can currently interact with map objects by tapping on them when using touch controls. This wouldn't work since tapping anywhere on the screen would just move the closest virtual input. Not much of an issue, because we could simply make it behave more like our physical gamepad behavior by adding a virtual button that maps to the "Accept" input.

I think it doesn't need to follow the user's touch i suggest the joystick just looks and feels like that in the video since movements might be a little more fluid when controlling the virtual joystick, But It'll be cool if the Basic attack button turns to action when near something instead of adding a third button for action, But either way sounds great!

**Comment by McSinyx** [comment]
> It'll be cool if the Basic attack button turns to action when near something instead of adding a third button for action

I don't think this will work for long-range attacks, which benefits more from an additional joystick.  I've been playing on the Pinephone for around an hour and it seems only the brute class is playable with current touch input thanks to the wide-angle attacks.  Long-range aiming is lacking precision since the movement joystick only has 8 directions.  In addition, that attacks blocking movements make it more difficult to aim.
